pytest_skip/plugin.py

```diff
@@ -16,6 +16,65 @@ class SelectOption(Enum):
     SKIP = "skipfromfile"
 
 
+for i, o in enumerate(SelectOption):
+    o.ordinal = i
+
+
+@dataclass
+class Matcher:
+    test_names: Dict[str, Union[True, Set[Pattern]]] = field(default_factory=dict)
+    seen_test_names: Set[str] = field(default_factory=set)
+
+    regexp_test_name_suffix: ClassVar[str] = "@regexp"
+    regexp_test_name_pattern: ClassVar[Pattern] = re.compile(r"^(.+)\[r\"(.*?)\"\]@regexp$")
+    test_name_pattern: ClassVar[Pattern] = re.compile(r"^(.+?)\[(.*)\]$")
+
+    def add(self, test_name: str):
+        # Check if the line has the regexp suffix before matching
+        # the whole pattern. Saves time on huge skip-lists
+        if test_name.endswith(self.regexp_test_name_suffix):
+            match = self.regexp_test_name_pattern.match(test_name)
+            if match is None:
+                warnings.warn(f"The line '{test_name}' has athe @regexp "
+                              "suffix but doesn't contain an actual regexp. The line will be "
+                              "treated as a non-regexp.")
+                self.test_names[test_name[:-len(self.regexp_test_name_suffix)]] = True
+            else:
+                test_name, regexp = match.groups()
+                regexps_list = self.test_names.get(test_name, set())
+                if regexps_list is True:
+                    return
```

Comment:
let's print a warning in that case, something like "the test {test_name} appears as a regexp and non-regexp, ignoring the regexp version"